src/modules/cpu/kernel/glitch.hpp

```diff
@@ -0,0 +1,1177 @@
+    }
+}
+
+RppStatus glitch_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                   RpptDescPtr srcDescPtr,
+                                   Rpp8u *dstPtr,
+                                   RpptDescPtr dstDescPtr,
+                                   Rpp32u *batch_x_offset_r,
+                                   Rpp32u *batch_y_offset_r,
+                                   Rpp32u *batch_x_offset_g,
+                                   Rpp32u *batch_y_offset_g,
+                                   Rpp32u *batch_x_offset_b,
+                                   Rpp32u *batch_y_offset_b,
+                                   RpptROIPtr roiTensorPtrSrc,
+                                   RpptRoiType roiType,
+                                   RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for (int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        __m128i maskR = _mm_setr_epi8(0, 0x80, 0x80, 3, 0x80, 0x80, 6, 0x80, 0x80, 9, 0x80, 0x80, 12, 0x80, 0x80, 15);
+        __m128i maskGB = _mm_setr_epi8(0x80, 1, 2, 0x80, 4, 5, 0x80, 7, 8, 0x80, 10, 11, 0x80, 13, 14, 0x80);
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u x_offset_r = batch_x_offset_r[batchCount];
+        Rpp32u y_offset_r = batch_y_offset_r[batchCount];
+        Rpp32u x_offset_g = batch_x_offset_g[batchCount];
+        Rpp32u y_offset_g = batch_y_offset_g[batchCount];
+        Rpp32u x_offset_b = batch_x_offset_b[batchCount];
+        Rpp32u y_offset_b = batch_y_offset_b[batchCount];
+
+        Rpp32u elementsInRowMax = srcDescPtr->w;
+
+        Rpp32u xOffsets[3] = {
+            x_offset_r,
+            x_offset_g,
+            x_offset_b};
+
+        Rpp32u yOffsets[3] = {
+            y_offset_r,
+            y_offset_g,
+            y_offset_b};
+
+        Rpp32u xOffsetsLoc[3] = {
+            x_offset_r,
+            x_offset_g,
+            x_offset_b};
+
+        Rpp32u yOffsetsLoc[3] = {
+            y_offset_r * elementsInRowMax,
+            y_offset_g * elementsInRowMax,
+            y_offset_b * elementsInRowMax};
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+        Rpp32u alignedLength = (bufferLength / 32) * 32;
+        Rpp32u alignedLengthPerChannel = (srcDescPtr->w / 16) * 16;
```

Comment:
Shouldn't this be 
    Rpp32u alignedLengthPerChannel = (roi.xywhROI.roiWidth / 16) * 16;
Please check